### PR TITLE
Display proper labels for Application data

### PR DESCRIPTION
**Pull request by ramonh**
This adds a constant to the `Application` model that defines the labels that will be displayed when viewing an `Application`.

The `Application#show` view will now display these labels.

This aims to address #206

### CODE REVIEW

File: app/models/application.rb [hunk 1]
```diff
@@ -18,6 +18,44 @@ class Application < ActiveRecord::Base
...
+        student1_banking_info: 'Banking Information (2nd student)',
+        student1_application_minimum_money: 'Minimum Money (2nd student)',
+        paid: 'Paid?',
+        rgsoc_only: 'Exclusively for RGSoC Alumni?',
+        misc_info: 'Additional information',
+        location: 'City'
+  }
+
+  def self.data_label(key)
+    APPLICATION_DATA_LABELS[key] ? APPLICATION_DATA_LABELS[key] : human_attribute_name(key)
```
Comment (carpodaster) [inline app/models/application.rb]: I would rather not copy the field translation copied to a constant (would require editing the semantics in two different places again). An idea would be to change the method body:

```ruby
def self.data_label(key)
  ApplicationDraft.human_attribute_name(key) ? ApplicationDraft.human_attribute_name(key) : human_attribute_name(key)
end
```
but unfortunately `human_attribute_name` always returns a value.

OTOH, we only need this in the iteration in ` app/views/applications/show.html.slim:38`, so technically we never access direct database column attibutes of `Application`. We could use that one:

```ruby
def self.data_label(key)
  ApplicationDraft.human_attribute_name(key)
end
```

What dou you think?

**Comment by ramonh** [comment]
Thanks for the feedback!

I'm not sure I understood correctly, but calling `human_attribute_name` seems to return the 'English-ified' key. So for example:
```ruby
ApplicationDraft.human_attribute_name(:misc_info)
```
would return 'Misc info', which is not quite what we want.

Have I misunderstood?

**Comment by ramonh** [comment]
Quick follow-up:

I've looked into it, and calling `human_attribute_name` should be returning the values from the locale file. I'll look into why this isn't happening.

**Comment by ramonh** [comment]
Ok now I got everything I need to get going. Thanks for clarifying, @carpodaster!

**Comment by carpodaster** [comment]
For "normal" columns / attributes, that's correct. It looks up in the locales files and if it doesn't find something specific it just calls `titleize` on the column name, I think.

However, all application draft attributes are saved in the `application_data` hash within the `Application` record. That hash what we're iterating over and what's displayed in the show view. Since it's hash-data we cannot use the locale file.

**Comment by ramonh** [comment]
Got it - Thanks @carpodaster! Will be closing this to rewrite things.

**Pull request closed without merging**